homeassistant/helpers/condition.py

```diff
@@ -615,6 +615,14 @@ def sun(
         condition_trace_set_result(False, message="no sunset today")
         return False
 
+    # Special case: before sunrise OR after sunset
+    if before == SUN_EVENT_SUNRISE and after == SUN_EVENT_SUNSET:
```

Comment:
In the case where there is no sunrise or sunset, I would expect this OR condition to still fire, so this should probably go before the two `if blah is Non` statements above.

Reply:
Yes, you are correct.